tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/ModifyRepositoryMetadataMojo.java

```diff
@@ -0,0 +1,151 @@
+     * Properties that don't exist in the repository are just silently ignored.
+     */
+    @Parameter(property = "p2.repository.properties.remove")
+    private List<String> propertiesToRemove = List.of();
+    /**
+     * The properties to add.<br>
+     * Already existing properties are overwritten and updated to the value specified.
+     */
+    @Parameter(property = "p2.repository.properties.add")
+    private Map<String, String> propertiesToAdd = Map.of();
+
+    /** Compress the repository index files {@code content.xml} and {@code artifacts.xml}. */
+    @Parameter(property = "p2.repository.compress", defaultValue = "true")
+    private boolean compress;
+
+    /**
+     * Add XZ-compressed repository index files. XZ offers better compression ratios esp. for highly
+     * redundant file content.
+     */
+    @Parameter(property = "p2.repository.xz", defaultValue = "true")
+    private boolean xzCompress;
+
+    /**
+     * If {@link #xzCompress} is {@code true}, whether jar or xml index files should be kept in
+     * addition to XZ-compressed index files. This fallback provides backwards compatibility for
+     * pre-Mars p2 clients which cannot read XZ-compressed index files.
+     */
+    @Parameter(property = "p2.repository.xz.keep", defaultValue = "true")
+    private boolean keepNonXzIndexFiles;
+
+    @Inject
+    private P2RepositoryDataManipulator manipulator;
+    @Inject
+    private Logger logger;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+        ModifiedRepositoryDescriptor descriptor = P2RepositoryDataManipulator.createDescriptor(repository,
+                repositoryKind, outputLocation, compress, xzCompress, keepNonXzIndexFiles);
+
+        String type = "repository";
+        if (repositoryKind != P2RepositoryKind.both) {
+            type = (repositoryKind.isArtifact() ? "artifact " : "metadata") + " repository";
+        }
+        String location = descriptor.sourceRepository().getUrl();
+        Path output = descriptor.outputLocation();
+        if (output != null) {
+            logger.info("  Write modified " + type + " file(s) to: " + output);
+        } else {
+            logger.info("  Modify " + type + " in-place at: " + location);
+        }
+        if (repositoryName != null) {
+            logger.info("  Set name to: " + repositoryName);
+        }
+        if (!propertiesToRemove.isEmpty()) {
+            logger.info("  Remove properties:" + String.join(", ", propertiesToRemove));
+        }
+        if (!propertiesToAdd.isEmpty()) {
+            logger.info("  Set properties:");
+            propertiesToAdd.forEach((key, value) -> {
+                logger.info("   " + key + " = " + value);
+            });
+        }
+        try {
+            manipulator.modifyRepositoryMetadata(descriptor, repositoryName, propertiesToRemove, propertiesToAdd);
+        } catch (ProvisionException | IOException | URISyntaxException e) {
```

Comment:
`ProvisionException` should be handled separately here by

getLog().error(StatusTool.toLogMessage(e.getStatus()), e);
throw new MojoExecutionException("Failed to modify repository " + location+": "+StatusTool.collectProblems(e.getStatus), e);

Reply:
Ok, done.